test/routes/site.route.test.js

```diff
@@ -110,8 +133,80 @@ lab.experiment('Site page tests:', () => {
       const parser = new DOMParser()
       const doc = parser.parseFromString(res.payload, 'text/html')
 
-      const element = doc.getElementById('site-error').firstChild
-      Code.expect(element.nodeValue).to.equal('ERROR: Invalid site name: [invalid_site_name]')
+      let element
+      let errorMessage = 'Enter the site name'
+
+      // Panel summary error item
+      element = doc.getElementById('error-summary-list-item-0').firstChild
+      Code.expect(element.nodeValue).to.equal(errorMessage)
+
+      // Site name field error
+      element = doc.getElementById('site-name-error').firstChild
+      Code.expect(element.nodeValue).to.equal(errorMessage)
+
+      done()
+    })
+  })
+
+  lab.test('POST /site shows an error message when the site name contains invalid characters', (done) => {
+    const request = {
+      method: 'POST',
+      url: '/site',
+      headers: {},
+      payload: {
+        siteName: '___INVALID_SITE_NAME___',
+        anotherName: ''
+      }
+    }
+
+    server.inject(request, (res) => {
+      Code.expect(res.statusCode).to.equal(200)
+
+      const parser = new DOMParser()
+      const doc = parser.parseFromString(res.payload, 'text/html')
+
+      let element
+      let errorMessage = 'The site name cannot contain any of these characters: ^ | _ ~ ¬ `'
```

Comment:
Can we get this text from the validator itself rather than copy it here? Otherwise we'll have to update this every time the microcopy or list of disallowed characters changes.

Reply:
As discussed we don't want to use the code to test itself in case the copy we would need to refer to inadvertently changes. We have agreed that for the time being in validation messages we will validate their copy in our unit tests as per this PR, but we won't try and test the remaining copy in the pages in our unit tests as we will leave that to our End to End tester and the page designers to sign off.